docs/rofl/features: Document public variables (#2487)
oasisprotocol/oasis-sdk: docs/rofl/features/public_var.md

# Public Variables

Sometimes containers need access to non-sensitive configuration, for example

> API endpoints or public feature flags. This data can be passed to containers

Reviewer: ```suggestion
API endpoints, contract addresses, feature flags, and other deliberately exposed information.
Such data can be passed to containers
```

running in ROFL via _public variables_. Public variables are arbitrary key-value
pairs that are exposed to containers as environment variables. Use them
for values that can (or should be) transparent to anyone.

Public variables can be easily managed via the Oasis CLI, for example to create
a public variable called `API_URL` you can use:

```sh
echo -n "https://api.example.com" | oasis rofl public-var set API_URL -
```

:::info Detailed CLI Reference

For comprehensive documentation on public variable management commands including
importing from `.env` files, removing public variables, and other advanced
features, consult the [Oasis CLI] documentation.

:::

> 

Reviewer: I'd move this info box somewhere at the end. Do the same in the Secrets chapter.

Note that this only updates the local app manifest file, but the public variable
is not propagated to the app just yet. This allows you to easily configure as
many public variables as you want without the need to constantly update the
on-chain app configuration.

> :::info

Reviewer: ```suggestion
:::info Public variables vs Secrets
```

Also update Secrets chapter to include reference to Public variables chapter.

Public variables are stored unencrypted in the local app manifest and in
on-chain app metadata. This means that they can be read by anyone who can access

> the manifest or inspect the on-chain app configuration.

Reviewer: ```suggestion
on-chain app metadata. Anyone who can access the manifest or inspect the on-chain app
configuration can read them.
```

Author: Removed this entirely as this is repeating the obvious once more, secrets/public var are references as alternative at the top of the chapter in both files.
